pkg/runtime/wfengine/workflow.go

```diff
@@ -298,6 +298,12 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 		}
 	}
 
+	// Increment the generation counter if the workflow used continue-as-new. Subsequent actions below
+	// will use this updated generation value for their duplication execution handling.
+	if runtimeState.ContinuedAsNew() {
+		state.Generation += 1
```

Comment:
I'm not super familiar with the flow of the code, but is this ever accessed by any other goroutine? (Maybe not yet, I see a comment "TODO: Parallelism"?) If it is, should use atomic.Add instead or this can cause race conditions

Reply:
No, this will never be accessed by any other goroutine. We can rely on actor concurrency guarantees for this. The TODO: Parallelism refers to parallel invocation of the "activity" actors further below, which won't modify `state`.